bids/layout/tests/test_models.py

```diff
@@ -38,6 +38,20 @@ def subject_entity():
                directory="{subject}", dtype='str')
 
 
+def test_layoutinfo_init():
+    args = dict(root='/made/up/path', validate=True,
+                      absolute_paths=True, index_metadata=False,
+                      derivatives=True, ignore=['code/', 'blergh/'],
+                      force_index=None)
+    with pytest.raises(ValueError) as exc:
+        LayoutInfo(**args)
+        assert exc.value.message.startswith("Missing mandatory")
```

Comment:
Exceptions pop out of the `pytest.raises()` context manager. To inspect `exc`'s contents, you need to to that in the regular scope.

```suggestion
    assert exc.value.message.startswith("Missing mandatory")
```